modules/report/view/saldobalance_finale.php

```diff
@@ -0,0 +1,103 @@
+<?
+// Fetch input parameters
+$enable_last_year = (bool) $_REQUEST['report_EnableLastYear'];
+$from_period = $_REQUEST['report_FromPeriod'];
+$to_period = $_REQUEST['report_ToPeriod'];
+$result_from_period = $_REQUEST['report_ResultFromPeriod'];
+
+list($report_year) = explode("-", $from_period);
+$company_name = $_lib['sess']->get_companydef('CompanyName');
+$last_year_text = ($enable_last_year) ? utf8_encode(' med fjoraarets') : '';
+$filename = $report_year .' '. $company_name .' Saldobalanse'. $last_year_text .'.csv';
+
+// Set header to a CSV file, and make it an attachment so it is downloaded and not displayed
+header('Content-Type: text/csv; charset=utf-8');
+header('Content-Disposition: attachment; filename="'. $filename .'"');
+
+// Open output stream as file output
+$output = fopen('php://output', 'w');
+
+$from_prev_period   = $_lib['date']->get_this_period_last_year($_REQUEST['report_FromPeriod'] . "-01");
+$to_prev_period     = $_lib['date']->get_this_period_last_year($_REQUEST['report_ToPeriod'] . "-01");
+$result_prev_from   = $_lib['date']->get_this_period_last_year($_REQUEST['report_ResultFromPeriod'] . "-01");
+
+$query_balance = "select A.AccountPlanID, A.AccountName from accountplan A where A.AccountPlanType = 'balance' or A.AccountPlanType = 'result' group by A.AccountPlanID order by A.AccountPlanID asc";
+$query_result  = str_replace("balance", "result", $query_balance);
+$balance_accounts = $_lib['db']->db_query($query_balance);
+$result_accounts = $_lib['db']->db_query($query_result);
+
+// Headings, left here just so we know what each column in CSV is
+// $headings = array('Account number', 'Account name', 'Amount');
+// if ($enable_last_year) array_push($headings, 'Amount last year');
+// fputcsv($output, $headings);
+
+// Loop over balance accounts
+while($account = $_lib['db']->db_fetch_object($balance_accounts))
+{
+    $query_saldo_old = "select sum(V.AmountIn) as sumin, sum(V.AmountOut) as sumout from voucher V where $VoucherType V.VoucherPeriod < '$from_period' and V.AccountPlanID='$account->AccountPlanID' and V.Active=1 group by V.AccountPlanID";
+    $voucher_saldo_old = $_lib['storage']->get_row(array('query' => $query_saldo_old));
+    $saldo_old      = (round($voucher_saldo_old->sumin, 2) - round($voucher_saldo_old->sumout, 2));
+
+    $query_saldo    = "select sum(V.AmountIn) as sumin, sum(V.AmountOut) as sumout from voucher V where $VoucherType V.VoucherPeriod >= '$from_period' and V.VoucherPeriod <= '$to_period' and V.AccountPlanID='$account->AccountPlanID' and V.Active=1 group by V.AccountPlanID";
+    $voucher_saldo  = $_lib['storage']->get_row(array('query' => $query_saldo));
+    $saldo_new_sum_in = round($voucher_saldo->sumin, 2);
+    $saldo_new_sum_out = round($voucher_saldo->sumout, 2);
+    $saldo_new      = $saldo_new_sum_in - $saldo_new_sum_out;
+
+    $sumrow_new     = $saldo_old + $saldo_new;

```

Comment:
Refactor the variable names to be more descriptive or add an example in comment what each is for